scripts/SHI/scpSHI_Jetz_complete.R

```diff
@@ -0,0 +1,396 @@
+#for this calculation and compares the use of different land cover sources
+#-------------------------------------------------------------------------------------------------------------------
+#0. Load libraries and define Parameters
+#-------------------------------------------------------------------------------------------------------------------
+# Script location can be used to access other scripts
+print(Sys.getenv("SCRIPT_LOCATION"))
+options(timeout = max(60000000, getOption("timeout")))
+
+## Install required packages
+packages <- c("devtools","rlang","dplyr","tidyr","ggplot2","remotes","purrr","tmap","raster","ggsci","readr",
+              "rgbif","rgdal","sf","httr","jsonlite","landscapemetrics","rjson","stars","geodata","rredlist",
+              "stacatalogue","gdalcubes","rstac","RColorBrewer","RCurl","tmaptools","gdalUtilities","OpenStreetMap")
+
+if (!"gdalcubes" %in% installed.packages()[,"Package"]) remotes::install_git("https://github.com/appelmar/gdalcubes_R.git", update="never")
+if (!"stacatalogue" %in% installed.packages()[,"Package"]) remotes::install_git("https://github.com/ReseauBiodiversiteQuebec/stac-catalogue", update="never", quiet=T)
+# 
+new.packages <- packages[!(packages %in% installed.packages()[,"Package"])]
+if(length(new.packages)) install.packages(new.packages)
+
+lapply(packages,require,character.only=T)
+
+path_script <- Sys.getenv("SCRIPT_LOCATION")
+source(file.path(path_script,"SHI/funGet_range_maps.R"), echo=TRUE)
+
+## Receiving args
+args <- commandArgs(trailingOnly=TRUE)
+outputFolder <- args[1] # Arg 1 is always the output folder
+cat(args, sep = "\n")
+
+input <- fromJSON(file=file.path(outputFolder, "input.json"))
+print("Inputs: ")
+print(input)
+
+#Define species
+sp <- input$species
+
+#Define CRS
+ref_system <- input$ref_system
+sf_crs <- st_crs(ref_system)
+srs_cube <- paste0("EPSG:",ref_system)
+
+#margin for elevation range
+elev_margin <- input$elev_margin
+#spatial resolution
+spat_res <- input$spat_res
+
+#define country if the area of analysis will be restricted to a specific country
+country_code <- ifelse(input$country_code== "" , NA,input$country_code)
+region <- ifelse(input$region== "" , NA ,input$region)
+
+#Define source of expert range maps
+expert_source <- input$expert_source
+#forest threshold for GFW (level of forest for the species)
+forest_threshold <- input$forest_threshold #USE MAP OF LIFE VALUES!!*****
+
+#define time steps
+t_0 <- input$t_0
+t_n <- input$t_n # should be larger than t_0 at least 2 years needed
+time_step <- input$time_step
+t_range <- ((t_n - t_0)/time_step)
+v_time_steps <- seq(t_0,t_n,time_step)
+
+#credentials
+token <- Sys.getenv("IUCN_TOKEN")
+
+#Filter source for land cover, global or colombia
+# LC_source <- input$lc_source
+
+## Receiving args
+args <- commandArgs(trailingOnly=TRUE)
+outputFolder <- args[1] # Arg 1 is always the output folder
+cat(args, sep = "\n")
+
+setwd(outputFolder)
+
```

Comment:
Consider a short description of the process here.

Reply:
added:
# Set output folder as working directory which has permissions to save files

Reply:
sorry for the size of the letter I added it outside the code chunk but with the # that gives a title